tests/python_package_test/test_dask.py

```diff
@@ -647,6 +647,98 @@ def test_ranker(output, client, listen_port, group):
+    if task == 'ranking':
+        X, y, w, g, dX, dy, dw, dg = _create_ranking_data(
+            output='dataframe',
+            group=group_sizes,
+            chunk_size=10
+        )
+        model_factory = lgb.DaskLGBMRanker
+        eval_metric=['ndcg']
+        eval_group = []
+    else:
+        X, y, w, dX, dy, dw = _create_data(
+            objective=task,
+            output='array',
+            chunk_size=10,
+        )
+        dg = None
+        eval_group = None
+        if task == 'classification':
+            model_factory = lgb.DaskLGBMClassifier
+            eval_metric = ['binary_error']
+        elif task == 'regression':
+            model_factory = lgb.DaskLGBMRegressor
+            eval_metric = ['l2']
+
+    full_trees = 200
+    params = {
+        "random_state": 42,
+        "n_estimators": full_trees,
+        "num_leaves": 31,
+        "min_child_samples": 1,
+        "verbose": 5,
+        "first_metric_only": True
+    }
+
+    dask_model = model_factory(
+        client=client
+        , **params
+    )
+
+    eval_set, eval_sample_weight = [], []
+    for eval_size in eval_sizes:
+        if eval_size == 1:
+            eval_set.append((dX, dy))
+            eval_sample_weight.append(dw)
+
+            if task == 'ranking':
+                eval_group.append(dg)
+        else:
+            eval_partitions = int(max(1, np.floor(eval_size * dX.npartitions)))
+            eval_set.append((dX.partitions[0:eval_partitions], dy.partitions[0:eval_partitions]))
+            eval_sample_weight.append(dw.partitions[0:eval_partitions])
+
+            if task == 'ranking':
+                eval_group.append(dg.partitions[0:eval_partitions])
+
+    # when eval_size is exactly 1 partition, not enough eval_set data to reach each worker.
+    if eval_sizes == [0]:
+        with pytest.raises(RuntimeError, match='eval_set was provided but worker .* was not allocated validation data'):
+            dask_model.fit(dX, dy, group=dg, eval_set=eval_set,
+                           eval_sample_weight=eval_sample_weight, eval_group=eval_group,
+                           eval_metric=eval_metric, early_stopping_rounds=5)
+
+    else:
+        dask_model = dask_model.fit(dX, dy, group=dg, eval_set=eval_set,
+                                    eval_sample_weight=eval_sample_weight, eval_group=eval_group,
+                                    eval_metric=eval_metric, early_stopping_rounds=5)
+        fitted_trees = dask_model.to_local().booster_.num_trees()
+        assert fitted_trees < full_trees
+
+        # be sure that model still produces decent output.
+        p1 = dask_model.predict(dX)
+        if task == 'classification':
+            print(f'_accuracy_score(dy, p1) = {_accuracy_score(dy, p1)}')
+            _accuracy_score(dy, p1) > 0.8
+
+        elif task == 'regression':
+            print(f'_r2_score(dy, p1) = {_r2_score(dy, p1)}')
+            _r2_score(dy, p1) > 0.8
+
+        else:
```

Comment:
Could you please add some additional checks on other attributes that will be set  when early stopping is enabled?

`best_score_` and `best_iteration_`? You can look in any of the sklearn tests that use `early_stopping_rounds` for reference: https://github.com/microsoft/LightGBM/blob/fd094d2719a04b984b376a665b6c3e5e1aff2967/tests/python_package_test/test_sklearn.py#L114

Reply:
Addressed in 1335591